peps/pep-0759.rst

```diff
@@ -0,0 +1,472 @@
+PEP: 759
+Title: External Wheel Hosting
+Author: Barry Warsaw <barry@python.org>,
+        Ethan Smith <ethan@ethanhs.me>
+BDFL-Delegate: Donald Stufft <donald@python.org>
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 19-Sep-2024
+Post-History:
+
+Abstract
+========
+
+This PEP proposes a mechanism by which projects hosted on `pypi.org
+<http://pypi.org>`__ can safely host wheel artifacts on external sites other
+than PyPI. This PEP explicitly does *not* propose external hosting of
+projects, packages, or their metadata. That functionality is already available
+by externally hosting independent package indexes. Because this PEP only
+provides a mechanism for projects to customize the download URL for specific
+released wheel artifacts, dependency resolution as already implemented by
+common installer tools such as `pip <https://pip.pypa.io/en/stable/>`__ and
+`uv <https://docs.astral.sh/uv/>`__ does not need to change.
+
+This PEP defines what it means to be "safe" in this context, along with a new
+package upload file format called a ``.rim`` file. It defines how ``.rim``
+files affect the metadata returned for a package's `Simple Repository API
+<https://packaging.python.org/en/latest/specifications/simple-repository-api/>`__
+in both HTML and JSON format, and how traditional wheels can easily be turned
+into ``.rim`` files.
+
+Rationale
+=========
+
+The Python Package Index, hosted at https://pypi.org, imposes `default limits
+<https://pypi.org/help/>`__ on upload artifact file size (100 MiB) and total
+project size (10.0 GiB). Most projects can comfortably fit within these limits
+during the lifetime of the project, through years of uploads. A few projects
+have encountered these limits, and have been granted both `file size
+exceptions <https://github.com/pypi/support/issues/4378>`__ and `project size
+exceptions <https://github.com/pypi/support/issues/4379>`__, allowing them to
```

Comment:
Is it worth linking to specific examples here? Slight concern that the links are more distracting than beneficial.

Reply:
Perhaps not.  Those are examples I've run into, but probably don't add much to the rationale.  I'll remove them.